services/QuillLMS/client/app/bundles/PremiumHub/containers/__tests__/skillSection.test.tsx

```diff
@@ -0,0 +1,56 @@
+import * as React from "react";
+import { render, screen } from "@testing-library/react";
+
+import SkillSection from "../diagnosticGrowthReports/skillSection";
+
+const props = {
+  loadingFilters: true,
+  customStartDate: null,
+  customEndDate: null,
+  pusherChannel: null,
+  searchCount: 0,
+  selectedClassrooms: [],
+  allClassrooms: [],
+  selectedGrades: [],
+  allGrades: [],
+  selectedSchools: [],
+  selectedTeachers: [],
+  allTeachers: [],
+  selectedTimeframe: {
+    label: "This school year",
+    name: "This school year",
+    value: "this-school-year"
+  },
+  handleClickDownloadReport: jest.fn(),
+  openMobileFilterMenu: jest.fn()
+}
+
+describe('SkillSection', () => {
+  describe('loading state', () => {
+    test('it should render loading spinner', () => {
+      const { asFragment } = render(<SkillSection {...props} />);
+      expect(asFragment()).toMatchSnapshot();
+      const loadingSpinner = screen.getByRole('img')
+      expect(loadingSpinner.getAttribute('class')).toEqual('spinner')
```

Comment:
This was just added, but as of [this change](https://github.com/empirical-org/Empirical-Core/blob/cea38a0bbad6826da3ddfcadb490ebb2434c381b/services/QuillLMS/client/app/bundles/Shared/components/shared/spinner.tsx), it's better to `queryByAltText` rather than look for attributes like class because those aren't visible to the user. So in this instance you would do something like: `expect(screen.getByAltText(/loading spinner/i)).toBeInTheDocument().`

Reply:
oh great, that's def cleaner-- thanks for flagging